wdl-lint/src/rules/inconsistent_newlines.rs

```diff
@@ -0,0 +1,83 @@
+//! A lint rule for ensuring that newlines are consistent.
+
+use wdl_ast::AstNode;
+use wdl_ast::Diagnostic;
+use wdl_ast::Diagnostics;
+use wdl_ast::Span;
+use wdl_ast::SyntaxKind;
+use wdl_ast::ToSpan;
+use wdl_ast::VisitReason;
+use wdl_ast::Visitor;
+
+use crate::Rule;
+use crate::Tag;
+use crate::TagSet;
+
+/// The identifier for the inconsistent newlines rule.
+const ID: &str = "InconsistentNewlines";
+
+/// Creates an inconsistent newlines diagnostic.
+fn inconsistent_newlines(span: Span) -> Diagnostic {
+    Diagnostic::note("inconsistent newlines detected")
+        .with_rule(ID)
+        .with_highlight(span)
```

Comment:
If we change `span` to be the span of the first mismatched newline, we should probably use a label to indicate that we're only showing the first such occurrence in the file:

```suggestion
        .with_label("the first mismatched newline is here", span)
```